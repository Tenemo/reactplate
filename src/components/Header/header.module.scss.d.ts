/* eslint-disable */
/* prettier-ignore */
// @ts-nocheck
/**
 * Generated by vite-css-modules
 * https://npmjs.com/vite-css-modules
 */

const themeLight: string;
const themeDark: string;
const header: string;
const themeToggle: string;
const gitHubLink: string;
const headerButtons: string;

export {
	themeLight,
	themeDark,
	header,
	themeToggle,
	gitHubLink,
	headerButtons
};

export default {
	"theme-light": themeLight,
	themeLight,
	"theme-dark": themeDark,
	themeDark,
	header,
	themeToggle,
	gitHubLink,
	headerButtons
};
