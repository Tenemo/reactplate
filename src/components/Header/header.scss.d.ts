// Automatically generated by @teamsupercell/typings-for-css-modules-loader.
// Please do not edit this file manually.
declare namespace HeaderScssNamespace {
  export interface IHeaderScss {
    header: string;
  }
}

declare const HeaderScssModule: HeaderScssNamespace.IHeaderScss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: HeaderScssNamespace.IHeaderScss;
};

export = HeaderScssModule;
