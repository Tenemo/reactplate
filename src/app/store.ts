import type { Action, ThunkAction } from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';
import { createLogger } from 'redux-logger';

import { appThemeSlice } from 'features/AppTheme/appThemeSlice';
import { exampleRequestSlice } from 'features/ExampleRequest/exampleRequestSlice';

export type RootState = ReturnType<typeof rootReducer>;

const logger = createLogger({
    diff: true,
    collapsed: true,
});

export const rootReducer = combineReducers({
    appTheme: appThemeSlice.reducer,
    exampleRequestApi: exampleRequestSlice.reducer,
});

const middleware = [exampleRequestSlice.middleware];

// We'll be just fine with inferring these types :)
// eslint-disable-next-line @typescript-eslint/explicit-function-return-type, @typescript-eslint/explicit-module-boundary-types
export const makeStoreDev = (preloadedState?: Partial<RootState>) => {
    const store = configureStore({
        reducer: rootReducer,
        devTools: true,
        middleware: (getDefaultMiddleware) =>
            getDefaultMiddleware().concat(middleware).concat(logger),
        preloadedState,
    });
    return store;
};

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type, @typescript-eslint/explicit-module-boundary-types
export const makeStoreProd = (preloadedState?: Partial<RootState>) => {
    const store = configureStore({
        reducer: rootReducer,
        devTools: false,
        middleware: (getDefaultMiddleware) =>
            getDefaultMiddleware().concat(middleware),
        preloadedState,
    });
    return store;
};
const buildType = import.meta.env.MODE;
export const makeStore =
    buildType === `development` ? makeStoreDev : makeStoreProd;

export const store = makeStore();

export type AppStore = typeof store;
export type AppDispatch = AppStore['dispatch'];
export type AppThunk<ThunkReturnType = void> = ThunkAction<
    ThunkReturnType,
    RootState,
    unknown,
    Action
>;
