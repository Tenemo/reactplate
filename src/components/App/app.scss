.app {
    @font-face {
        font-family: "Roboto Mono";
        src: url("../../fonts/RobotoMono-Regular.woff2") format("woff2"),
            url("../../fonts/RobotoMono-Regular.woff") format("woff");
    }
    font-family: "Roboto Mono", "Courier New", "Courier", monospace;
    padding: 10px 15px;
    height: 100%;
    display: flex;
    flex-direction: column;
}
:global .theme-dark {
    color: #ddd;
    background-color: #222;
}
:global .theme-light {
    color: #222;
    background-color: #eee;
}
.error {
    color: darkred;
    margin: 10px 20px;
}
.errorInfo {
    color: #aaa;
    margin-top: 20px;
}
