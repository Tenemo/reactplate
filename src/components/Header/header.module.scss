@use "styles/variables";

.header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    gap: 1rem;
    padding: 0.5rem 1rem;
    path {
        fill: var(--color-text);
    }
    a {
        text-decoration: none;
    }
    margin-bottom: 1rem;
    border-bottom: 1px solid var(--color-text);
}
.themeToggle {
    cursor: pointer;
    &:hover path {
        fill: var(--color-link-hover) !important;
    }

    &:visited path {
        fill: var(--color-text);
    }
}
.gitHubLink {
    display: flex;
    align-items: center;
    gap: 0.5rem;
    text-decoration: none;
    &:hover path {
        fill: var(--color-link-hover) !important;
    }

    &:visited path {
        fill: var(--color-text);
    }
}

.headerButtons {
    display: flex;
    flex-direction: row;
    gap: 1rem;
    align-items: flex-start;
}
