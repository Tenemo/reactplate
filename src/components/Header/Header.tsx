import React, { ReactElement } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import { getAppTheme } from 'store/app/appSelectors';
import { toggleTheme } from 'store/app/appActions';

import styles from './header.scss';

export const Header = (): ReactElement => {
    const dispatch = useDispatch();
    const appTheme = useSelector(getAppTheme);

    const onToggleThemeClick = (): void => {
        dispatch(toggleTheme());
    };

    return (
        <header className={styles.header}>
            <button onClick={onToggleThemeClick} type="button">
                Change to {appTheme === 'dark' ? 'light' : 'dark'}
            </button>
        </header>
    );
};

export default Header;
