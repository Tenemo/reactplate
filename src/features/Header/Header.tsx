import React from 'react';

import styles from './header.module.scss';

import { useAppSelector, useAppDispatch } from 'app/hooks';
import { toggleTheme, selectAppTheme } from 'features/AppTheme/appThemeSlice';

const Header = (): React.JSX.Element => {
    const dispatch = useAppDispatch();
    const appTheme = useAppSelector(selectAppTheme);

    const onToggleThemeClick = (): void => {
        dispatch(toggleTheme());
    };

    return (
        <header className={styles.header}>
            <button onClick={onToggleThemeClick} type="button">
                Change to {appTheme === 'dark' ? 'light' : 'dark'}
            </button>
        </header>
    );
};

export default Header;
