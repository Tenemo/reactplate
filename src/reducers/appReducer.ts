import * as types from 'constants/actionTypes';
import { Action, AppStore } from 'store';

export const initialState: AppStore = {
    theme: `dark`,
    requestsInProgress: 0,
};

const actionTypeIsRequestEnd = (type: types.ActionType): boolean =>
    type.substring(type.length - 8) === `_SUCCESS` ||
    type.substring(type.length - 8) === `_FAILURE`;

const actionTypeIsRequest = (type: types.ActionType): boolean =>
    type.substring(type.length - 8) === `_REQUEST`;

const listReducer = (
    state: AppStore = initialState,
    action: Action,
): AppStore => {
    if (actionTypeIsRequest(action.type)) {
        return { ...state, requestsInProgress: state.requestsInProgress + 1 };
    }
    if (actionTypeIsRequestEnd(action.type)) {
        return { ...state, requestsInProgress: state.requestsInProgress - 1 };
    }
    return state;
};

export default listReducer;
