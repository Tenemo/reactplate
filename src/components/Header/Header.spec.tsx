import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import createMockStore, { MockStore } from 'redux-mock-store';

import { APP_TOGGLE_THEME } from 'store/app/appTypes';
import { initialAppState } from 'store/app/appReducer';
import { RootState } from 'store/types';

import { Header } from './Header';

let mockStore: MockStore;

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn((selector: (state: RootState) => void) =>
        selector(mockStore.getState()),
    ),
    useDispatch: jest.fn(() => mockStore.dispatch),
}));

jest.mock('store/app/appActions', () => {
    return {
        toggleTheme: jest.fn(() => ({
            type: 'APP_TOGGLE_THEME',
        })),
    };
});

describe('Header', () => {
    const emptyMockStore = createMockStore([thunk]);

    let reactRedux: typeof import('react-redux');
    type MockUseSelector = typeof reactRedux.useSelector & {
        mockClear: () => void;
    };
    type MockUseDispatch = typeof reactRedux.useDispatch & {
        mockClear: () => void;
    };

    let appActions: typeof import('store/app/appActions');
    type MockAppActions = {
        toggleTheme: {
            mockClear: () => void;
        };
    };

    beforeAll(async () => {
        reactRedux = await import('react-redux');
        appActions = await import('store/app/appActions');
    });
    beforeEach(() => {
        mockStore = emptyMockStore({ app: initialAppState });
    });
    afterEach(() => {
        (reactRedux.useSelector as MockUseSelector).mockClear();
        (reactRedux.useDispatch as MockUseDispatch).mockClear();
        (appActions as unknown as MockAppActions).toggleTheme.mockClear();
    });

    it('should match the snapshot', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper).toMatchSnapshot();
    });
    it('clicking theme toggle button should change the theme', () => {
        const wrapper = mount(<Header />);
        const ToggleThemeButton = wrapper.find('button');
        ToggleThemeButton.simulate('click', {
            preventDefault: () => undefined,
        });
        expect(mockStore.getActions()).toEqual([{ type: APP_TOGGLE_THEME }]);
        expect(appActions.toggleTheme).toHaveBeenCalledTimes(1);
    });
});
