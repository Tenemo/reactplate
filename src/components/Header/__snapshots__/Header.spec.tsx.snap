// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<header>
  <button
    onClick={[Function]}
    type="button"
  >
    Change to 
    light
  </button>
</header>
`;
